src/openforms/submissions/logic/rules.py

```diff
@@ -141,7 +141,9 @@ def iter_evaluate_rules(
             span_type="app.submissions.logic",
             labels={"ruleId": rule.pk},
         ):
-            triggered = bool(jsonLogic(rule.json_logic_trigger, data_container.data))
+            triggered = bool(
+                evaluate_json_logic(rule.json_logic_trigger, data_container.data, rule)
+            )
```

Comment:
I'm a bit partial about the evaluation actually happening in the `log_utils` module.

I think we can have a nicer api using a context manager (`@contextlib.contextmanager`), something like:

```py
with log_evaluation(rule, rule.json_logic_trigger):
    triggered = bool(jsonLogic(rule.json_logic_trigger, data_container.data))
```

and 

```py
with log_evaluation(rule, action_details["value"]):
    new_value = jsonLogic(action_details["value"], data_container.data))
```